src/msg/async/EventKqueue.cc

```diff
@@ -156,8 +156,13 @@ int KqueueDriver::add_event(int fd, int cur_mask, int add_mask)
     }
   }
   // keep what we set
-  if (fd >= sav_max)
-    resize_events(sav_max+5000);
+  if (fd >= sav_max) {
+    r = resize_events(sav_max+5000);
+    if (r < 0) {
+      lderr(cct) << __func__ << " event count is exceed." << dendl;
+      return r;
```

Comment:
```c++
  if (r < 0) {
    // Actually we don't allow any failed error code, caller doesn't prepare to
    // handle error status. So now we need to assert failure here. In practice,
    // add_event shouldn't report error, otherwise it must be a innermost bug!
    assert(0 == "BUG!");
    return r;
  }
```

one of its caller. i still prefer "crash early".